modules/ROOT/pages/push/handling-push-notifications.adoc

```diff
@@ -1,3 +1,62 @@
 include::{partialsdir}/attributes.adoc[]
 
 = Handling Push Notifications
+
+Follow the next steps to handle incoming {push-notification}s in your foregrounded application.
+
+NOTE: {push-notification}s that arrive when the application is in the background will always be handled by the OS.
+
+[role="primary"]
+.Android
+****
+Add your implementation of *MessageHandler* to your application's manifest.
+
+[source,xml]
+----
+<meta-data
+    android:name="DEFAULT_MESSAGE_HANDLER_KEY"
+    android:value=".MyMessageHandler" />
+----
+****
+
+
+[role="secondary"]
+.iOS
```

Comment:
@jhellar could you evaluate if this is correct? I could use some help here.

Reply:
I think that you first need to handle remote notification with https://github.com/aerogear/ios-showcase-template/blob/master/ios-showcase-template/AppDelegate.swift#L116. NotificationCenter is not part of AGSPush. It can be used to post the notification (https://github.com/aerogear/ios-showcase-template/blob/master/ios-showcase-template/push/PushHelper.swift#L18-L20) to a View so that it can be rendered. @wtrocki can you please confirm?

Reply:
I agree with @wtrocki  that this is too big to take on, and we should skip it.. 
However, I disagree with the general point. It's ok to document any steps that user needs to perform in order to be successful with mobile services regardless of software origin...  provided that documentation task is reasonable (and in this case, I don't think it is)

Reply:
@finp problem here is that this code will outdate faster than yoghurt in the fridge and people will actually get better experience if we point them to: 
https://developer.apple.com/notifications/